src/main/java/pason/storage/Storage.java

```diff
@@ -0,0 +1,145 @@
+ */
+public class Storage {
+    private static final String FILE_DIRECTORY = "data";
+    private static final String FILE_NAME = "tasks.txt";
+
+    /**
+     * Initialises the Storage class.
+     */
+    public Storage() {
+    }
+
+    /**
+     * Loads stored tasks from file.
+     * If valid tasks exist, add to tasks.
+     *
+     * @throws FileNotFoundException  If file directory or name is not found
+     * @throws PasonException  If the task list contains incorrect format
+     */
+    public List<Task> loadTasks() throws FileNotFoundException, PasonException {
+        File directory = new File(FILE_DIRECTORY);
+        File file = new File(FILE_DIRECTORY + "/" + FILE_NAME);
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                throw new PasonException(e.getMessage());
+            }
+        }
+        List<Task> tasks = new ArrayList<>();
+        Scanner s = new Scanner(file);
+        Task task;
+        int failedImports = 0;
+        while (s.hasNext()) {
+            task = parseFileEntry(s.nextLine());
+            if (task != null) {
+                tasks.add(task);
+            } else {
+                failedImports++;
+            }
+        }
+        if (failedImports > 0) {
+            throw new PasonException("There was a problem parsing "
+                    + failedImports + " task(s) from the file.");
+        }
+        return tasks;
+    }
+    /**
+     * Parses the given task entry String.
+     * Returns null if given string is in incorrect format.
+     *
+     * @param text  Text entry to be parsed.
+     * @return Task object parsed from text.
+     */
+    public static Task parseFileEntry(String text) {
+        String[] splitString = text.split(" \\| ");
+        if (splitString[0].equals("T") && splitString.length == 3) {
+            ToDo newToDo = new ToDo(splitString[2]);
+            if (splitString[1].equals("1")) {
+                newToDo.markAsDone();
+            }
+            return newToDo;
+        } else if (splitString[0].equals("E") && splitString.length == 4) {
+            String[] eventDate = splitString[3].split(" ");
+            Event newEvent = new Event(splitString[2],
+                    LocalDate.parse(eventDate[0]), (eventDate.length == 1 ? null : eventDate[1]));
+            if (splitString[1].equals("1")) {
+                newEvent.markAsDone();
+            }
+            return newEvent;
+        } else if (splitString[0].equals("D") && splitString.length == 4) {
+            Deadline newDeadline = new Deadline(splitString[2],
+                    LocalDateTime.parse(splitString[3]));
+            if (splitString[1].equals("1")) {
+                newDeadline.markAsDone();
+            }
+            return newDeadline;
+        } else {
+            return null;
+        }
```

Comment:
I like how neat this if else block is. There seem to be some hardcoded strings passed into the split function - perhaps they could be assigned to some variables with descriptive names, or other comments could be added to explain more about what's happening in this function?